### PR TITLE
[WFCORE-820] Attribute for-hosts for Security Realm with Kerberos doe…

### PR DESCRIPTION
**Pull request by TomasHofman**
…s not select keytab in defined order

https://bugzilla.redhat.com/show_bug.cgi?id=1228228
https://issues.jboss.org/browse/WFCORE-820

### CODE REVIEW

**Comment by TomasHofman** [comment]
Retest this please.

File: server/src/main/resources/schema/jboss-as-config_1_7.xsd [hunk 1]
```diff
@@ -461,8 +461,8 @@
                             
                             At the time authentication is going to be handled the keytab will be selected as follows: -
                              1 - Iterate the list of keytabs and identity one where the for-hosts attribute contains an entry matching protocol/hostname.
-                             2 - Iterate the list of keytabs and identity one where the for-hosts attribute contains an entry matching hostname.
-                             3 - Iterate the list of keytabs and identify one where the name of the principal matches matches protocol/hostname.
+                             2 - Iterate the list of keytabs and identify one where the name of the principal matches matches protocol/hostname.
+                             3 - Iterate the list of keytabs and identity one where the for-hosts attribute contains an entry matching hostname.
                              4 - Iterate the list of keytabs and identify one where the hostname portion of the principal matches the hostname of the request.
                              5 - Use the keytab where for-hosts is set to '*'.
```
Comment (bstansberry) [inline server/src/main/resources/schema/jboss-as-config_1_7.xsd]: The current schema is https://github.com/wildfly/wildfly-core/blob/master/server/src/main/resources/schema/wildfly-config_4_0.xsd

**Comment by TomasHofman** [comment]
Ah, thanks Brian. Didn't realize there may be newer version. So probably only fix the latest one...?

Can I actually change XSD documentation just like that? Should I involve documentation team for some reason?

**Comment by bstansberry** [comment]
Hi Tomas,

Sorry, I didn't check back to see your comment.

Generally we don't update schemas once they are published in a final release. It's confusing to have different variations of the same schema out there on the internet. So just the latest is ok.

If you need to fix BZ 1228228 (currently it's not triaged for inclusion in a CP and I doubt it ever will be) and someone insists on updating the 1.7 schema as the fix, we won't object to changing it in WildFly Core as well.

Re: docs, probably the simplest thing to do is change the "Doc Type" on the BZ to Known Issue and fill in the Doc Text. I believe that will provide the info needed to create an entry in the EAP Release Notes if docs wants one.

**Comment by bstansberry** [comment]
retest this please

**Pull request merged**